### DISCUSSION TITLE
Does isort support my approach?

### DISCUSSION THREAD

**Discussion by href** [PyCQA/isort#2225] [Q&A]
Hi!

I've been writing Python for over a decade and I've ultimately arrived at this approach that has not changed in a few years. From what I can tell this can't be done with isort, but maybe I didn't find the right settings:

This is how my organisation typically looks like in my code:
```python
from __future__ import annotations

import os
import requests

from typing import Any
from typing import Dict
```

- The first block contains all `__future__` imports, sorted by name.
- The second block contains all `import` directives, sorted by name.
- The third block contains all `from` directives, sorted by name.

I do not use any commas in my imports, that means I never do `from foo import bar, baz`. I've stopped doing that at some point as it makes for easier sorting and nicer Git histories.

Basically what I do is write my import in the right block, then sort the block.

Any way isort can support this style? Thanks for letting me know.

**Comment by mshafer-NI** [comment]
Not sure about the blocks..., but you can force isort to keep each import on separate lines with `force_single_line=true` \[[ref](https://pycqa.github.io/isort/docs/configuration/options.html#force-single-line)\]

**Comment by guillermodotn** [comment]
Have a look at the option `sort_order`.
